[Bug]:  unexpected server crashed by uncatched runtime panic(aoe unsupported /tae npe)

### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```


### Actual Behavior

AOE doesn't support `delete`, but it's better to catch this error to avoid a server crash.

```
panic: implement me

goroutine 800 [running]:
github.com/matrixorigin/matrixone/pkg/vm/engine/aoe/engine.(*relation).GetPrimaryKeys(0xc000027fa0?, {0xc0265e9e8c?, 0x2?, 0x2?})
	/mnt/datadisk0/tmp/0602/matrixone/pkg/vm/engine/aoe/engine/relation.go:257 +0x27
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnCompilerContext).GetPrimaryKeyDef(0xc003528240, {0xc023cb3d5d?, 0xc0033e5860?}, {0xc0265e9e8c, 0x2})
	/mnt/datadisk0/tmp/0602/matrixone/pkg/frontend/session.go:670 +0x39f
github.com/matrixorigin/matrixone/pkg/sql/plan2.buildDelete(0xc010b5fea0, {0x30e5d58?, 0xc003528240})
	/mnt/datadisk0/tmp/0602/matrixone/pkg/sql/plan2/build_delete.go:65 +0x287
github.com/matrixorigin/matrixone/pkg/sql/plan2.BuildPlan({0x30e5d58?, 0xc003528240?}, {0x30cde28?, 0xc010b5fea0?})
	/mnt/datadisk0/tmp/0602/matrixone/pkg/sql/plan2/build.go:49 +0x3f7
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnComputationWrapper).Compile(0xc022d0df80, {0x10?, 0x1?}, 0xc00312ea80?)
	/mnt/datadisk0/tmp/0602/matrixone/pkg/frontend/mysql_cmd_executor.go:1527 +0x51
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery(0xc0037c9770, {0xc0265e9e80, 0x10})
	/mnt/datadisk0/tmp/0602/matrixone/pkg/frontend/mysql_cmd_executor.go:1820 +0x1862
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest(0xc0037c9770, 0xc000027ee0)
	/mnt/datadisk0/tmp/0602/matrixone/pkg/frontend/mysql_cmd_executor.go:2081 +0x5d3
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop(0xc008ca3740)
	/mnt/datadisk0/tmp/0602/matrixone/pkg/frontend/routine.go:106 +0x264
created by github.com/matrixorigin/matrixone/pkg/frontend.NewRoutine
	/mnt/datadisk0/tmp/0602/matrixone/pkg/frontend/routine.go:158 +0x1e5

```

TAE seems to be unable to support deletion as well, when using TAE+plan2, `delete` query will cause panic error as well.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x16fb979]

goroutine 234 [running]:
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/moengine.(*txnRelation).GetPrimaryKeys(0x2ddaea2?, {0xc00059ad8c?, 0xc001e05310?, 0x2?})
	/home/runner/work/matrixone/matrixone/head/pkg/vm/engine/tae/moengine/relation.go:93 +0x79
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnCompilerContext).GetPrimaryKeyDef(0xc0004fa330, {0xc001d837b5?, 0xc001e05430?}, {0xc00059ad8c, 0x2})
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/session.go:670 +0x39f
github.com/matrixorigin/matrixone/pkg/sql/plan2.buildDelete(0xc00ed7c0f0, {0x315c738?, 0xc0004fa330})
	/home/runner/work/matrixone/matrixone/head/pkg/sql/plan2/build_delete.go:65 +0x287
github.com/matrixorigin/matrixone/pkg/sql/plan2.BuildPlan({0x315c738?, 0xc0004fa330?}, {0x31445a8?, 0xc00ed7c0f0?})
	/home/runner/work/matrixone/matrixone/head/pkg/sql/plan2/build.go:49 +0x3f7
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnComputationWrapper).Compile(0xc00ed3b470, {0x0?, 0x1?}, 0xc002047fb0?)
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:1527 +0x51
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery(0xc0000ad900, {0xc00059ad80, 0x2d})
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:1820 +0x1862
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest(0xc0000ad900, 0xc00ed1f040)
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:2081 +0x5d3
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop(0xc0000161e0)
		/home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:106 +0x264
created by github.com/matrixorigin/matrixone/pkg/frontend.NewRoutine
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:158 +0x1e5

```

### Expected Behavior

_No response_

### Steps to Reproduce

_No response_

### Additional information

_No response_

--- a/pkg/frontend/mysql_cmd_executor.go
+++ b/pkg/frontend/mysql_cmd_executor.go
@@ -24,6 +24,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/matrixorigin/matrixone/pkg/common/moerr"
 	"github.com/matrixorigin/matrixone/pkg/errno"
 	"github.com/matrixorigin/matrixone/pkg/sql/compile2"
 	"github.com/matrixorigin/matrixone/pkg/sql/errors"
@@ -2034,8 +2035,14 @@ func (mce *MysqlCmdExecutor) handleDDl(ses *Session, stmt tree.Statement, epoch 
 }
 
 // ExecRequest the server execute the commands from the client following the mysql's routine
-func (mce *MysqlCmdExecutor) ExecRequest(req *Request) (*Response, error) {
-	var resp *Response = nil
+func (mce *MysqlCmdExecutor) ExecRequest(req *Request) (resp *Response, err error) {
+	defer func() {
+		if e := recover(); e != nil {
+			err = moerr.NewPanicError(e)
+			resp = NewGeneralErrorResponse(COM_QUERY, err)
+		}
+	}()
+
 	logutil.Infof("cmd %v", req.GetCmd())
 
 	ses := mce.GetSession()

